common/CMakeLists.txt

```diff
@@ -4,6 +4,23 @@ find_package(Threads REQUIRED)
 
 llama_add_compile_flags()
 
+if (NOT MSVC)
+    if (LLAMA_SANITIZE_THREAD)
+        add_compile_options(-fsanitize=thread)
+        link_libraries     (-fsanitize=thread)
+    endif()
+
+    if (LLAMA_SANITIZE_ADDRESS)
+        add_compile_options(-fsanitize=address -fno-omit-frame-pointer)
+        link_libraries     (-fsanitize=address)
+    endif()
+
+    if (LLAMA_SANITIZE_UNDEFINED)
+        add_compile_options(-fsanitize=undefined)
+        link_libraries     (-fsanitize=undefined)
+    endif()
+endif()
```

Comment:
If this is done in the top-level `CMakeLists.txt` it should automatically apply to all included subdirectories.

Reply:
I moved them to the `llama_add_compile_flags()` to be consistent with the existing `LLAMA_FATAL_WARNINGS` and `LLAMA_ALL_WARNINGS` options.

Moved a single call of `llama_add_compile_flags()` to the top-level `CMakeLists.txt`. It's OK for now, but in general I think that having separate calls in the sub-trees is better, because it allows for example to have 3rd-party code in the `examples` tree that would not be a subject to the llama.cpp compile flags.

Reply:
The sanitizer flags should still be applied to 3rd-party code regardless, but maybe not other flags like warnings.